modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMFormValuesToFieldsConverterImpl.java

```diff
@@ -252,4 +257,23 @@ private void _addMissingRepeatedDDMFormFieldValues(
 		}
 	}
 
+	private void _filterDDMFormFieldValues(
+		Map<String, DDMFormField> ddmFormFieldsMap,
+		List<DDMFormFieldValue> ddmFormFieldValues) {
+
+		for (DDMFormFieldValue ddmFormFieldValue : ddmFormFieldValues) {
+			List<DDMFormFieldValue> nestedDDMFormFieldValues =
+				ddmFormFieldValue.getNestedDDMFormFieldValues();
+
+			if (!nestedDDMFormFieldValues.isEmpty()) {
+				_filterDDMFormFieldValues(
+					ddmFormFieldsMap, nestedDDMFormFieldValues);
+			}
+
+			ddmFormFieldValues.removeIf(
+				ddmFormFieldValue1 -> !ddmFormFieldsMap.containsKey(
```

Comment:
Hey, @marcelabc. I know you don't like to change the pull request when CI's results pass, so I'm gonna give my notes right now, in case it saves some time: the `ddmFormFieldValue1` should be 2 and the first `ddmFormFieldValue` should be `ddmFormFieldValue1`

Thanks a lot.

Reply:
thank you!